core/state/state_object.go

```diff
@@ -523,6 +523,10 @@ func (s *stateObject) Balance() *big.Int {
 	return s.data.Balance
 }
 
+func (s *stateObject) Root() common.Hash {
```

Comment:
You're only accessing this from package `state`, so it doesn't have to be public
```suggestion
func (s *stateObject) root() common.Hash {
```

Reply:
Alternatively, keep `Root` public, and change the whole thing into
```golang
	if !state.Exist(address) {
		return nil, nil
	}
	obj := state.GetOrNewStateObject(addr)
	return &types.StateAccount{
		Nonce:    obj.Nonce(),
		Balance:  obj.Balance(),
		Root:     obj.Root(),
		CodeHash: obj.CodeHash(),
	}, state.Error()
}
```